java/src/main/java/com/microsoft/tunnels/websocket/WebSocketSession.java

```diff
@@ -72,6 +74,8 @@ public void channelInactive(ChannelHandlerContext ctx) throws Exception {
 
   @Override
   public void channelRead(ChannelHandlerContext ctx, Object msg) throws Exception {
+    this.reading.acquire();
+    this.reading.release();
```

Comment:
Should the release happen after `super.channelRead()`?

Reply:
Multiple reads are allowed at the same time, we just need to check if reads are blocked before starting